lxd/container_lxc.go

```diff
@@ -5299,24 +5315,20 @@ func (c *containerLXC) fillNetworkDevice(name string, m shared.Device) (shared.D
 				return nil, err
 			}
 
-			c.localConfig[configKey] = volatileName
-			c.expandedConfig[configKey] = volatileName
-
 			// Update the database
-			tx, err := dbBegin(c.daemon.db)
-			if err != nil {
-				return nil, err
-			}
-
-			err = dbContainerConfigInsert(tx, c.id, map[string]string{configKey: volatileName})
+			err = updateKey(configKey, volatileName)
 			if err != nil {
-				tx.Rollback()
-				return nil, err
-			}
+				// Check if something else filled it in behind our back
+				value, err1 := dbContainerConfigGet(c.daemon.db, c.id, configKey)
+				if err1 != nil || value == "" {
+					return nil, err
```

Comment:
Not sure I follow. We were setting up a separate DB connection for name and hwaddr, so the two were separate operations, failure of updating the hwaddr wouldn't cause the name to be reverted for example.

So in that regard I don't think anything changed. The only change is that now, when something fails we go check if that's because someone filled the missing information behind our back, if so, we set the appropriate fields to that value and move on without failing.